src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs

```diff
@@ -628,7 +628,7 @@ private static ICalibratorTrainer GetCalibratorTrainerOrThrow(IExceptionContext
         /// <param name="useProbabilities">Use probabilities (vs. raw outputs) to identify top-score category.</param>
         /// <typeparam name="TModel">The type of the model. This type parameter will usually be inferred automatically from <paramref name="binaryEstimator"/>.</typeparam>
         public static OneVersusAllTrainer OneVersusAll<TModel>(this MulticlassClassificationCatalog.MulticlassClassificationTrainers catalog,
-            ITrainerEstimator<ISingleFeaturePredictionTransformer<TModel>, TModel> binaryEstimator,
+            ITrainerEstimator<BinaryPredictionTransformer<TModel>, TModel> binaryEstimator,
```

Comment:
>BinaryPredictionTransformer [](start = 30, length = 27)

Might not be so easy because any given `BinaryPredictionTransformer` might not be an `ISingleFeaturePredictionTransformer`; from what I understand, `OVA` is currently limited to learners with single feature vectors (e.g. not `FieldAwareFactorizationMachines`).

Reply:
Heer is the definition for BinaryPredictionTransformer:

    /// <summary>
    /// Base class for the <see cref="ISingleFeaturePredictionTransformer{TModel}"/> working on binary classification tasks.
    /// </summary>
    /// <typeparam name="TModel">An implementation of the <see cref="IPredictorProducing{TResult}"/></typeparam>
    public sealed class BinaryPredictionTransformer<TModel> : SingleFeaturePredictionTransformerBase<TModel>


---
In reply to: [265408746](https://github.com/dotnet/machinelearning/pull/2949#discussion_r265408746) [](ancestors = 265408746)